lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -332,19 +323,55 @@ def run_batch(
         :param as_admin: Allows you to run jobs in other people's workspaces
         :return: A list of condor job ids or a failure notification
         """
-        wsid = batch_params.get("wsid")
+        if type(params) != list:
+            raise IncorrectParamsException("params must be a list")
+        wsid = batch_params.get(_WORKSPACE_ID)
         meta = batch_params.get("meta")
         if as_admin:
             self.sdkmr.check_as_admin(requested_perm=JobPermissions.WRITE)
         else:
             # Make sure you aren't running a job in someone elses workspace
             self._check_workspace_permissions(wsid)
-            wsids = [job_input.get("wsid", wsid) for job_input in params]
+            # this is very odd. Why check the parent wsid again if there's no wsid in the job?
+            # also, what if the parent wsid is None?
+            # also also, why not just put all the wsids in one list and make one ws call?
+            wsids = [job_input.get(_WORKSPACE_ID, wsid) for job_input in params]
```

Comment:
Seems like an oversight? Could probably just be collapsed into a single call to `_check_workspace_permissions_list`. I get it's out of scope for this PR, but maybe make a ticket?

Reply:
https://github.com/kbase/execution_engine2/issues/351